common4j/src/main/com/microsoft/identity/common/java/crypto/key/AbstractAES256SecretKeyProvider.java

```diff
@@ -22,42 +22,21 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.java.crypto.key;
 
-import lombok.NonNull;
-
-public abstract class AES256KeyLoader extends AbstractSecretKeyLoader {
-    private static final String TAG = AES256KeyLoader.class.getSimpleName();
 
+/**
+ * Abstract base class for loading AES-256 secret keys.
+ * <p>
+ * This class implements the {@link ISecretKeyProvider} interface and provides a default
+ * implementation for retrieving an AES-256 key generator. It serves as a base class
+ * for concrete implementations that handle the loading of AES-256 secret keys from
+ * various sources.
+ */
+public abstract class AbstractAES256SecretKeyProvider implements ISecretKeyProvider {
```

Comment:
doesn't look like this abstract class is overriding anything. do we need this?

if all we need are constants/static object - we can get rid of this class and
- expose KEY_ALGORITHM in AES256SecretKeyGenerator
- add a static getInstance() in AES256SecretKeyGenerator() to replace AES_256_KEY_GENERATOR